nnels_solr_search/nnels_solr_search.module

```diff
@@ -39,36 +53,45 @@ function nnels_solr_search_entity_property_info_alter(&$info) {
-  if ($file_resources) {
+  if (variable_get('nnels_certified_accessibility_taxonomy', FALSE)) {
+    $file_resources = field_get_items('node', $item, 'field_file_resource');
+    if ($file_resources) {
 
-    // Load all terms with the required certified field checked. The
-    // 'Accessibility Tags' Taxonomy has a field called 'Required for
-    // Certification' that tells which terms control the certified accessible
-    // icon.
-    $certified_accessibility_tags = db_select('field_data_field_required_for_certification', 'r')
-      ->fields('r', ['entity_id'])
-      ->condition('r.field_required_for_certification_value', 1)
-      ->groupBy('r.entity_id')
-      ->execute()
-      ->fetchCol('entity_id');
+      // Load all terms with the required certified field checked. The
+      // 'Accessibility Tags' Taxonomy has a field called 'Required for
+      // Certification' that tells which terms control the certified accessible
+      // icon.
+      $certified_accessibility_tags = db_select('field_data_field_required_for_certification', 'r')
+        ->fields('r', ['entity_id'])
+        ->condition('r.field_required_for_certification_value', 1)
+        ->groupBy('r.entity_id')
+        ->execute()
+        ->fetchCol('entity_id');
+      if (count($certified_accessibility_tags)) {
 
-    $file_resource_ids = array();
-    foreach ($file_resources as $file_resource) {
-      $file_resource_ids[$file_resource['value']] = TRUE;
-    }
-    $file_resource_entities = field_collection_item_load_multiple(array_keys($file_resource_ids));
-    foreach ($file_resource_entities as $file_resource_entity) {
-      $accessibility_tags = field_get_items('field_collection_item', $file_resource_entity, 'field_accessibility_tags');
-      if ($accessibility_tags) {
-        $accessibility_tags = array_column($accessibility_tags, 'tid');
+        $file_resource_ids = array();
+        foreach ($file_resources as $file_resource) {
+          $file_resource_ids[$file_resource['value']] = TRUE;
+        }
+        $file_resource_entities = field_collection_item_load_multiple(array_keys($file_resource_ids));
+        foreach ($file_resource_entities as $file_resource_entity) {
+          $accessibility_tags = field_get_items('field_collection_item', $file_resource_entity, 'field_accessibility_tags');
+          if ($accessibility_tags) {
+            $accessibility_tags = array_column($accessibility_tags, 'tid');
 
-        // Expecting $diff to be empty if all of the required tags are present
-        // in the attached tags.
-        $diff = array_diff($certified_accessibility_tags, $accessibility_tags);
-        if (empty($diff)) {
-          return 1;
+            // Expecting $diff to be empty if all of the required tags are present
+            // in the attached tags.
+            $diff = array_diff($certified_accessibility_tags, $accessibility_tags);
+            if (empty($diff)) {
+              return 1;
```

Comment:
I see the getter callback is type integer, but does it function as a flag, i.e. boolean? if so should return TRUE.

Reply:
It actually can return any value. And the value that is returned will be indexed and later displayed in the facet. I'm re-writing the facet item later on so 1 = "NNELS Certified". I figured int would be better for indexing/searching...but I don't know this for sure.

Reply:
Ah as key:value, yes that is ideal for indexing.

I retract the nitpick - approving 👍